crates/ignore/src/gitignore.rs

```diff
@@ -250,7 +251,12 @@ impl Gitignore {
         if self.is_empty() {
             return Match::None;
         }
-        let path = path.as_ref();
+        let path_buf: PathBuf = if path.as_ref().to_str().unwrap().is_empty() {
+            env::current_dir().unwrap()
+        } else {
+            path.as_ref().to_path_buf()
```

Comment:
This is doing an alloc and a copy just to make the types unify. And this is called for _every_ match. Absolutely cannot do that.